.circleci/config.yml.j2

```diff
@@ -171,30 +175,63 @@ workflows:
               only:
                 - master
                 - slack-build-approvals
-{%- endfor %}
-{%- endfor %}
-{%- if image_map.keys() | dev_releases | length > 0 %}
+      {%- endfor %}{# distribution in distributions #}
+      {%- endif %}{# not edge_build #}
+      {%- endfor %}{# ac_version, distributions in image_map.items() #}
+  {%- if image_map.keys() | dev_releases | length > 0 %}
   nightly:
     when:
       and:
         - equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
         - equal: [ "every-midnight-utc", << pipeline.schedule.name >> ]
     jobs:
-{%- for ac_version, distributions in image_map.items() %}
-{%- set airflow_version = ac_version | get_airflow_version %}
-{%- if "dev" in ac_version and airflow_version not in dev_allowlist  %}
-{%- for distribution in distributions %}
+      {%- for ac_version, distributions in image_map.items() %}
+      {%- set airflow_version = ac_version | get_airflow_version %}
+      {%- set edge_build = ac_version | is_edge_build -%}
+      {%- if "dev" in ac_version and airflow_version not in dev_allowlist  %}
+
+      {%- if edge_build %}
+      - download-file:
+          name: download-latest-{{ airflow_version | replace('.dev', '') }}-build-info
+          url: https://pip.astronomer.io/simple/astronomer-certified/latest-{{ airflow_version | replace('.dev', '') }}.build.json
+          file: latest-{{ airflow_version | replace('.dev', '') }}.build.json
+      - write-version-to-tag-file:
+          name: write-version-to-tag-file-{{ airflow_version | replace('.dev', '') }}
+          input_json_file: latest-{{ airflow_version | replace('.dev', '') }}.build.json
+          output_file: extra-tags-{{ airflow_version | replace('.dev', '') }}.txt
+          requires:
+            - download-latest-{{ airflow_version | replace('.dev', '') }}-build-info
```

Comment:
nit

we can probably add a `{%- set airflow_version_wout_dev = airflow_version | replace('.dev', '') -%}`

as `{{ airflow_version | replace('.dev', '')` is repeated number of times

Reply:
Fixed.